corehq/apps/api/odata/views.py

```diff
@@ -141,6 +142,46 @@ def get(self, request, domain):
         return add_odata_headers(HttpResponse(metadata, content_type='application/xml'))
 
 
+class ODataFormServiceView(View):
+
+    urlname = 'odata_form_service_from_export_instance'
+
+    @method_decorator(basic_auth_or_try_api_key_auth)
+    @method_decorator(require_permission(Permissions.edit_data, login_decorator=None))
+    @method_decorator(toggles.ODATA.required_decorator())
+    def get(self, request, domain):
+        service_document_content = {
+            '@odata.context': absolute_reverse(ODataFormMetadataView.urlname, args=[domain]),
+            'value': [
+                {
+                    'name': config.get_id,
+                    'kind': 'EntitySet',
+                    'url': config.get_id,
+                }
+                for config in get_odata_form_configs_by_domain(domain)
+            ]
+        }
+        return add_odata_headers(JsonResponse(service_document_content))
+
+
+class ODataFormMetadataView(View):
+
+    urlname = 'odata_form_metadata_from_export_instance'
+
+    @method_decorator(basic_auth_or_try_api_key_auth)
+    @method_decorator(require_permission(Permissions.edit_data, login_decorator=None))
+    @method_decorator(toggles.ODATA.required_decorator())
+    def get(self, request, domain):
+        configs = get_odata_form_configs_by_domain(domain)
+        config_ids_to_properties = OrderedDict()
+        for config in sorted(configs, key=lambda _config: _config.get_id):  # For deterministic tests
```

Comment:
Is it possible for the sorting to happen in the tests instead of here?

Reply:
good point: https://github.com/dimagi/commcare-hq/pull/24770